dolosse/generic_json_consumer.py

```diff
@@ -0,0 +1,153 @@
+            valid = False
+        if not valid:
+            try:
+                following = int(decode[other_key], 16) #Catch hexadecimals
+                valid = True
+            except (ValueError, TypeError):
+                valid = False
+        if isinstance(decode[other_key], dict):
+            populate(variables,
+                     decode[other_key],
+                     count,
+                     name+": ")
+        else:
+            if name in variables:
+                if valid:
+                    variables[name].append(following)
+                else:
+                    variables[name].append(
+                        variables[name][-1])
+            else:
+                variables[name] = [0]*(count-1)
+                if valid:
+                    variables[name].append(following)
+                else:
+                    variables[name].append(0)
+                    #Remember, the list might still be empty here
+
+
+def get_args():
+    """Creates and returns a kafka consumer"""
+    parser = ArgumentParser()
+    parser.add_argument("--topic", default="default",
+                        help="The name of the kafka topic from which to read")
+    parser.add_argument("--group", default="my-group",
+                        help="The kafka group in which the topic exists")
+    parser.add_argument("--server", default="localhost:9092",
+                        help="Name and port of the kafka server (e.g. localhost:9092)")
+    arguments = parser.parse_args()
+
+    # To consume latest messages and auto-commit offsets
+    return arguments.topic, arguments.group, arguments.server
+
+def redraw(figures):
+    """Redraw all figures from a dict of figures"""
+    for key in figures:
+        figures[key].canvas.draw()
+        figures[key].canvas.flush_events()
+
+
+def update_data(data, value):
+    """Update the list of data, given some json input"""
+    try:
+        update = loads(value)
+    except decoder.JSONDecodeError as problem:
+        print("Error: Flawed JSON in value: ",
+              value)
+        print("Error message: ", problem)
+        update = ""
+    data.update(update)
+
+
+def graph(topic, group, server):
+    """Main function - reads json data from a kafka topic and graphs it"""
+
+    screen = initscr()
+    cbreak()
+    screen.nodelay(True)
+    screen.getch()
+
+    print("Generic Graphing Consumer")
+    print("Press any key to exit...")
+
+    data = {}
+    line = {}
+    plots = {}
+    figures = {}
+    axes = {}
+    count = 0
+    pyplot.ion() #Interactive mode on
+    configuration = {"bootstrap.servers": server,
+                     "group.id": group,
+                     "default.topic.config" : {
+                         "auto.offset.reset": "earliest"},
+                     "enable.auto.commit": False}
+    consumer = Consumer(configuration)
+    consumer.subscribe([topic])
+    keyboard_input = ERR
+
+    while True:
+        if keyboard_input != ERR:
```

Comment:
If you instantiate `keyboard_input` to None you can just use `if not keyboard_input`, which reads cleaner. And removes an import. Alternatively, you can use a try/catch to stop the program with Control+C by catching `KeyboardInterrupt`

Reply:
The screen.getch() function returns ERR, not None, if there has been no keyboard input to capture. "if not keyboard_input" would thus exit immediately.

Reply:
As I mentioned, I wouldn't expect it to return none, but ERR is a misnomer here. It's not really an error. If not keyboard input wouldn't exit immediately because of the `not`. It will run as long as `keyboard_input` is none. As soon as it has a value it'd exit.

Reply:
I agree that it's a misnomer, so I changed the code to remove the misnomer (using "from... import.. as..." to rename it to something a little better named).

Reply:
That's fine.